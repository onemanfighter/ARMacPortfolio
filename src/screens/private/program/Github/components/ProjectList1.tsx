import { GithubState, appStore, githubSelector } from '@appStore';
import { Box, Button, ButtonGroup } from '@chakra-ui/react';
import { useShallow } from 'zustand/react/shallow';

const projectList = [
  {
    name: 'OneManFighter',
    project: 'onemanfighter',
  },
  {
    name: 'Ar Mac Portfolio',
    project: 'ar-mac-portfolio',
  },
];

const ProjectList1 = () => {
  const { setProject } = appStore(useShallow(githubSelector));

  const handleProject = (project: string) => {
    setProject(project as GithubState['project']);
  };

  return (
    <Box
      display={'flow'}
      flexDir={'column'}
      scrollBehavior={'smooth'}
      padding={2}
      style={{ overflowY: 'scroll', height: '100%' }}
    >
      {projectList.map((data) => (
        <Button
          key={data.project}
          padding={2}
          margin={1}
          onClick={() => {
            handleProject(data.project);
          }}
        >
          {data.name}
        </Button>
      ))}
    </Box>
  );
};

export default ProjectList1;
