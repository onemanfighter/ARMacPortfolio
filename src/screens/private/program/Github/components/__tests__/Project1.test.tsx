import { fireEvent, render, screen } from '@testing-library/react';
import ProjectList1 from '../ProjectList1';

describe('ProjectList1', () => {
  it('should return default ProjectList1', () => {
    const { container } = render(<ProjectList1 />);

    expect(container).toMatchSnapshot();
  });

  it('should invoke callback on button press', () => {
    const { container } = render(<ProjectList1 />);

    fireEvent.click(screen.getByText('Ar Mac Portfolio'));

    expect(container).toMatchSnapshot();
  });
});
