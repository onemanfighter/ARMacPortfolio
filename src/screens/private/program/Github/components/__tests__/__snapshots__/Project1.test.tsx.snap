// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectList1 should invoke callback on button press 1`] = `
<div>
  <div
    class="css-1v7y4j0"
    style="overflow-y: scroll; height: 100%;"
  >
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      OneManFighter
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      Ar Mac Portfolio
    </button>
  </div>
</div>
`;

exports[`ProjectList1 should return default ProjectList1 1`] = `
<div>
  <div
    class="css-1v7y4j0"
    style="overflow-y: scroll; height: 100%;"
  >
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      OneManFighter
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      Ar Mac Portfolio
    </button>
  </div>
</div>
`;
